SPEC.md

```diff
@@ -64,50 +64,55 @@ It includes a *rootfs* with all of the files that will exist in the root of the
 
 ### Image Archives
 
-The ACI archive format aims for flexibility and relies on very boring technologies: HTTP, gpg, tar and gzip.
-This set of formats makes it easy to build, host and secure a container using technologies that are battle-tested.
+The ACI file format ("image archive") aims for flexibility and relies on standard and common technologies: HTTP, gpg, tar and gzip.
+This set of formats makes it easy to build, host and secure a container using technologies that are widely available and battle-tested.
 
-Image archives MUST be a tar formatted file.
-The image may be optionally compressed with gzip, bzip2, or xz.
-After compression, images may also be encrypted with AES symmetric encryption.
+- Image archives MUST be a tar formatted file.
+- All files in the image MUST maintain all of their original properties, including timestamps, Unix modes, and extended attributes (xattrs).
+- Image archives MUST be named with the suffix `.aci`, irrespective of compression/encryption (see below).
+- Image archives SHOULD be signed using PGP, in detached signature mode.
```

Comment:
Note that the current implementation in Rocket expects _late signing_ - e.g. the etcd ACI is signed *after* compression. This is contrary to what was implicitly described here (in the example commands below), where the signature is taken of the canonical image format (uncompressed, unencrypted tar) - which I am now making explicit here.

I have been going back and forth on this a little bit since there are some arguments both ways.

Signing the final (compressed/encrypted ACI) means that signature validation can happen as early as possible after transport and before any other processing, and also means that a `.sig` sitting next to an `.aci` can always be expected to match, which is a pretty nice property. On the other hand, since we would generally advise implementations to store the canonical image, it means that signature validation is probably _only_ useful at transport time (otherwise implementations would need to store information about compression/encryption and then reconstruct it later, ugh). I am actually OK with this (limiting signature validation to a property of the retrieval process), since we always have the cryptographic image ID to verify contents anyway at any other time.

Signing the uncompressed tar means that it is super easy to reproduce and the same signature can be used across a variety of transport formats. However, then the filenames are decoupled and working with ACIs and signatures on the command line becomes a total pain since you always have to check what you are working with before being able to check the signature. (Possible solution to this would be to introduce different file extensions to represent the different forms). It also means greater computational expense of having to always decrypt/decompress before verifying signatures. (Arguably this last point is just a short-term overhead since we will eventually be exploring streaming formats anyway).